File: app/src/main/java/com/classicsviewer/app/data/ReferencesManifest.kt
package com.classicsviewer.app.data

import org.json.JSONObject

data class ReferenceEntry(
    val id: String,
    val filename: String,
    val title: String,
    val author: String,
    val language: String,
    val pageCount: Int,
    val sizeBytes: Long,
)

data class ReferencesManifest(
    val version: Int,
    val entries: List<ReferenceEntry>,
) {
    companion object {
        fun parse(json: String): ReferencesManifest {
            val obj = JSONObject(json)
            val arr = obj.getJSONArray("entries")
            val list = ArrayList<ReferenceEntry>(arr.length())
            for (i in 0 until arr.length()) {
                val e = arr.getJSONObject(i)
                list.add(
                    ReferenceEntry(
                        id = e.getString("id"),
                        filename = e.getString("filename"),
                        title = e.getString("title"),
                        author = e.getString("author"),
                        language = e.getString("language"),
                        pageCount = e.getInt("pageCount"),
                        sizeBytes = e.getLong("sizeBytes"),
                    )
                )
            }
            return ReferencesManifest(obj.getInt("version"), list)
        }
    }
}

File: ios/ClassicsViewer/Models/ReferencesManifest.swift
import Foundation

/// One reference work shipped in the References ODR pack.
/// Mirrors Android `ReferenceEntry` and the on-disk `references_manifest.json`.
struct ReferenceEntry: Codable, Identifiable, Equatable {
    let id: String
    let filename: String
    let title: String
    let author: String
    let language: String
    let pageCount: Int
    let sizeBytes: Int64
}

/// Top-level structure of `references_manifest.json`.
struct ReferencesManifest: Codable, Equatable {
    let version: Int
    let entries: [ReferenceEntry]
}
